runhouse/resources/folders/folder.py

```diff
@@ -682,34 +722,36 @@ def rm(self, contents: list = None, recursive: bool = True):
         Example:
             >>> my_folder.rm()
         """
-        folder_path = Path(self.path).expanduser()
-
-        if contents:
-            for content in contents:
-                content_path = folder_path / content
-                if content_path.exists():
-                    if content_path.is_file():
-                        content_path.unlink()
-                    elif content_path.is_dir() and recursive:
-                        shutil.rmtree(content_path)
-                    else:
-                        raise ValueError(
-                            f"Path {content_path} is a directory and recursive is set to False"
-                        )
+        if self._use_http_endpoint:
+            self.system._rm(self.path, contents, recursive)
         else:
-            if recursive:
-                shutil.rmtree(folder_path)
-            else:
-                if folder_path.is_dir():
-                    for item in folder_path.iterdir():
-                        if item.is_file():
-                            item.unlink()
+            folder_path = Path(self.path).expanduser()
+            if contents:
```

Comment:
this is the same logic that's in the folder helper methods? can we consolidate or call those from here?